pkg/controller/kubelet-config/kubelet_config_controller.go

```diff
@@ -221,11 +221,20 @@ func (ctrl *Controller) cascadeDelete(cfg *mcfgv1.KubeletConfig) error {
 	if len(cfg.GetFinalizers()) == 0 {
 		return nil
 	}
-	mcName := cfg.GetFinalizers()[0]
-	err := ctrl.client.MachineconfigurationV1().MachineConfigs().Delete(mcName, &metav1.DeleteOptions{})
-	if err != nil && !macherrors.IsNotFound(err) {
+	finalizerName := cfg.GetFinalizers()[0]
+	mcs, err := ctrl.client.MachineconfigurationV1().MachineConfigs().List(metav1.ListOptions{})
+	if err != nil {
 		return err
 	}
+	for _, mc := range mcs.Items {
+		if string(mc.GetUID()) == finalizerName || mc.GetName() == finalizerName {
```

Comment:
Clarification question: I read this as:

    If the UID or the Name is the finalizerName then delete the MC by it's Name.

Is there any _reasonable_ possibility that the name _is_ the UID but us *not* the Name?

Reply:
My guess is the answer is no, but I'd like to verify :slightly_smiling_face:

Reply:
No.... The delete API always uses the name. :smiley: